libs/model/src/services/evmChainEvents/chain-event-utils.ts

```diff
@@ -60,24 +64,65 @@ const referralNamespaceDeployedMapper: EvmMapper<
     data: event.rawLog.data,
     topics: event.rawLog.topics,
   });
-  return {
+  return Promise.resolve({
     event_name: 'NamespaceDeployedWithReferral',
     event_payload: {
       ...event,
       parsedArgs: decoded.args,
     },
-  };
+  });
 };
 
-const launchpadTokenCreatedMapper: EvmMapper<'LaunchpadTokenCreated'> = (
+const launchpadTokenCreatedMapper: EvmMapper<'LaunchpadTokenCreated'> = async (
```

Comment:
I don't think this approach is the correct way to go. It adds a lot of room for issues in a place where we don't have retries. If we start adding arbitrary logic to chain-events service we will start to face issues where we are missing events. We haven't done it so far so that we can take advantage of queues and the natural retry logic that exists there. Chain-events  will never skip an event but with this introduction, it may and likely will. If we want to fully enrich events then they should go through a separate queue before being routed to downstream consumers.

Reply:
Chain events should be about 1 thing and 1 thing only: fetching events. It shouldn't be about enriching events and the complexity that that introduces (performance, error retrying, etc).